csclient/csclient.go

```diff
@@ -531,3 +548,25 @@ func (cs *Client) Log(typ params.LogType, level params.LogLevel, message string,
 	resp.Body.Close()
 	return nil
 }
+
+// Login explicitly obtains authorization credentials
+// for the charm store and stores them in the client's
+// cookie jar.
+func (cs *Client) Login() error {
```

Comment:
It would be nice if it looked for an existing cookie first, so that it doesn't login unnecessarily.